autorag/nodes/queryexpansion/run.py

```diff
@@ -0,0 +1,174 @@
+
+
+def run_query_expansion_node(modules: List[Callable],
+                             module_params: List[Dict],
+                             previous_result: pd.DataFrame,
+                             node_line_dir: str,
+                             strategies: Dict,
+                             ) -> pd.DataFrame:
+    """
+    Run evaluation and select the best module among query expansion node results.
+
+    :param modules: Query expansion modules to run.
+    :param module_params: Query expansion module parameters.
+    :param previous_result: Previous result dataframe.
+        In this case, it would be qa data.
+    :param node_line_dir: This node line's directory.
+    :param strategies: Strategies for query expansion node.
+    :return: The best result dataframe.
+    """
+    if not os.path.exists(node_line_dir):
+        os.makedirs(node_line_dir)
+    project_dir = pathlib.PurePath(node_line_dir).parent.parent
+    resources_dir = os.path.join(project_dir, "resources")
+    data_dir = os.path.join(project_dir, "data")
+    retrieval_gt = pd.read_parquet(os.path.join(project_dir, "data", "qa.parquet"))['retrieval_gt'].tolist()
+
+    # run query expansion
+    result_queries, execution_times = zip(*map(lambda task: measure_speed(
+        task[0], project_dir=project_dir, previous_result=previous_result, **task[1]), zip(modules, module_params)))
+    average_times = list(map(lambda x: x / len(result_queries[0]), execution_times))
+
+    # pop top_k from strategies
+    if "top_k" in strategies.keys():
+        top_k = strategies.pop("top_k")
+    else:
+        top_k = 10  # default value
+
+    # get retrieval module values
+    if "retrieval_module" in strategies.keys():
+        retrieval_modules = strategies.pop("retrieval_module")
+    else:
+        retrieval_modules = [{"module_type": "bm25"}]  # default value
+
+    # get the best retrieval result for each module
+    results = list(map(lambda x: module_best_retrieval(x, top_k, retrieval_modules, resources_dir,
+                                                       retrieval_gt, data_dir, strategies), result_queries))
+
+    # save results to folder
+    save_dir = os.path.join(node_line_dir, "query_expansion")  # node name
+    if not os.path.exists(save_dir):
+        os.makedirs(save_dir)
+    filepaths = list(map(lambda x: os.path.join(save_dir, make_module_file_name(x[0].__name__, x[1])),
+                         zip(modules, module_params)))
+    list(map(lambda x: x[0].to_parquet(x[1], index=False), zip(results, filepaths)))  # execute save to parquet
+    filenames = list(map(lambda x: os.path.basename(x), filepaths))
+
+    summary_df = pd.DataFrame({
+        'filename': filenames,
+        'module_name': list(map(lambda module: module.__name__, modules)),
+        'module_params': module_params,
+        'execution_time': average_times,
+        **{metric: list(map(lambda result: result[metric].mean(), results)) for metric in strategies.get('metrics')},
```

Comment:
The name of each column of metrics will be different.
Suggest you set prefix to every column name. Like 'query_expansion_retrieval_f1'